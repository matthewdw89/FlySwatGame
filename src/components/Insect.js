import React, { Component } from 'react';

const Insect = ({alive, hit, id, style}) => {

  let bugImg= alive ? 'alive' : 'dead'; 
  return (
      <div onClick={() => { hit(id) }} style={style} id={bugImg}>
          <img src={alive ? './images/bug2.png' : './images/bug2Dead.svg'} alt="Bug"/>
      </div>
  )
};

export default Insect;
