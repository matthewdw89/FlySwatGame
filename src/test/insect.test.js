import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Insect from '../components/Insect';

configure({ adapter: new Adapter() });

describe('<Insect>', () => {

  it('should render correctly', () => {
    const component = shallow(<Insect />);
    expect(component).toMatchSnapshot();
  });

  it('should register a click', () => {
    let clickFn = jest.fn();
    const component = shallow(<Insect hit={clickFn}/>);
    component.simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });

  it('should have a different id when dead', () => {
    let alive = false;
    const component = shallow(<Insect alive={alive} />);
    expect(component.find('#dead').exists()).toEqual(true);
  });

  it('should have a different image when dead', () => {
    let alive = false;
    const component = shallow(<Insect alive={alive} />);
    console.log(component.debug())
    expect(component.find('img').prop('src')).toEqual('./images/bug2Dead.svg')
  })

});