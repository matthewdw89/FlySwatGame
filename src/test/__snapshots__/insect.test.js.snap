// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Insect> should render correctly 1`] = `
<div
  id="dead"
  onClick={[Function]}
>
  <img
    alt="Bug"
    src="./images/bug2Dead.svg"
  />
</div>
`;
